components/core/tests/test-NetworkReader.cpp

```diff
@@ -193,29 +193,39 @@ TEST_CASE("network_reader_illegal_offset", "[NetworkReader]") {
 }
 
 TEST_CASE("network_reader_with_valid_http_header_kv_pairs", "[NetworkReader]") {
+    constexpr size_t cNumMaxTrials{10};
     std::unordered_map<std::string, std::string> valid_http_header_kv_pairs;
     // We use httpbin (https://httpbin.org/) to test the user-specified headers. On success, it is
     // supposed to respond all the user-specified headers as key-value pairs in JSON form.
-    constexpr int cNumHttpHeaderKeyValuePairs{10};
+    constexpr size_t cNumHttpHeaderKeyValuePairs{10};
     for (size_t i{0}; i < cNumHttpHeaderKeyValuePairs; ++i) {
         valid_http_header_kv_pairs.emplace(
                 fmt::format("Unit-Test-Key{}", i),
                 fmt::format("Unit-Test-Value{}", i)
         );
     }
-    clp::NetworkReader reader{
-            "https://httpbin.org/headers",
-            0,
-            false,
-            clp::CurlDownloadHandler::cDefaultOverallTimeout,
-            clp::CurlDownloadHandler::cDefaultConnectionTimeout,
-            clp::NetworkReader::cDefaultBufferPoolSize,
-            clp::NetworkReader::cDefaultBufferSize,
-            valid_http_header_kv_pairs
-    };
-    auto const content{get_content(reader)};
-    REQUIRE(assert_curl_error_code(CURLE_OK, reader));
-    auto const parsed_content = nlohmann::json::parse(content);
+    std::optional<std::vector<char>> content;
```

Comment:
```suggestion
    std::optional<std::vector<char>> optional_content;
```
Missed in the previous review, mb